### PR TITLE
Fix 'smudged' files after attributes are disabled

### PR DESCRIPTION
**Pull request by kjedruczyk**
Files in our working copy might have been 'smudged' by some
filters. After repo-wide attributes override is written those
smudged files might be considered as 'modified' because
apropriate clean filter is no longer applied.

To fix that side effect we need to rebuild the working copy
after the attributes are modified.

Test plan:

```
$ testbed/arcyd/test_shell.sh
$ arcyd stop # in arcyd working dir
$ (cd var/repos/localhost_fs1_origin &&
      echo '*.id ident' > .gitattributes &&
      echo '$Id' > test.id &&
      git add test.id && git commit -m "ident test")
# optionally modify repository working copy to verify that simply disabling attributes doesn't have side effects
$ arcyd fsck --fix
```

### CODE REVIEW

File: py/phl/phlgitx_ignoreattributes.py [hunk 1]
```diff
@@ -61,8 +62,23 @@ def ensure_repo_ignoring(repo_path):
...
+    repo = phlsys_git.Repo(repo_path)
     repo_attributes_path = os.path.join(repo_path, _REPO_ATTRIBUTES_PATH)
 
+    # Files in our working copy might have been 'smudged' by some
+    # filters. After repo-wide attributes override is written those
+    # smudged files might be considered as 'modified' because
+    # apropriate clean filter is no longer applied.
+    #
+    # To fix that side effect we need to rebuild the working copy
+    # after the attributes are modified.
```
Comment (aevri) [inline py/phl/phlgitx_ignoreattributes.py]: Good comment! :)

**Comment by aevri** [comment]
It feels slightly complicated and risky to me to use stash to save and restore state without user interaction. For example, if there are any merge conflicts on restoring then we'd complicate things for the user. If there are errors when writing the new attributes then we must take care to restore the stashed changes, otherwise it may be surprising that they're gone.

To make things simpler, it could be an error if this is called when the working copy is dirty - we could raise in that case. It would be the responsibility of the caller to save/restore state if necessary. Does that sound good?

**Comment by cs-shadow** [comment]
I agree with @aevri here that we probably should abort if working copy is dirty - saving to stash doesn't feels like the right thing to do. Also, if the command is interrupted after the changes were saved but before they were restored, they would leave the working copy in slightly weird state and user wouldn't know about the stash.

**Comment by aevri** [comment]
Thanks!

**Pull request merged**